luavm/src/lib/instructions/functions.rs

```diff
@@ -10,6 +11,28 @@ pub fn closure(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
         .bytecode
         .get_function(vm.curr_func)
         .get_func_index(second_arg(instr) as usize);
-    vm.registers[first_arg(instr) as usize] = LuaVal::from(LuaClosure::new(index));
+    vm.registers.last_mut().unwrap()[first_arg(instr) as usize] =
+        LuaVal::from(LuaClosure::new(index));
+    Ok(())
+}
+
+pub fn call(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
+    // The closure_index method gives us an index of the bytecode.functions vector
+    // where we have to "jump" in order to find the instructions of the callee.
+    let index = vm.registers.last().unwrap()[first_arg(instr) as usize].closure_index()?;
+    let old_func = vm.curr_func;
+    vm.curr_func = index;
+    // generate a new set of registers for the new stack frame
+    let reg_num = vm.bytecode.get_function(vm.curr_func).reg_count();
+    let mut new_regs = Vec::with_capacity(reg_num);
```

Comment:
This doesn't feel right. Consider a real processor: it has (say) 16 global registers. When a function calls another function, the latter stores registers it will mess about with on the stack and restores them when it returns. This minimises the amount of allocation and to-ing and fro-ing needed.

Reply:
Yes, that makes more sense, and should be a lot faster than my current implementation!